gapic/src/main/com/google/gapid/views/PipelineView.java

```diff
@@ -184,202 +192,293 @@ protected void onUiThreadError(Loadable.Message error) {
   protected void setPipelines(List<API.Pipeline> pipelines) {
     loading.stopLoading();
     disposeAllChildren(stagesContainer);
-    TabFolder folder = createStandardTabFolder(stagesContainer);
-    createPipelineTabs(folder, pipelines);
+    createPipelineTabs(stagesContainer, pipelines);
     stagesContainer.requestLayout();
   }
 
-  private void createPipelineTabs(TabFolder folder, List<API.Pipeline> pipelines) {
+  private void createPipelineTabs(Composite folder, List<API.Pipeline> pipelines) {
+    HashMap<String, Composite> stageMap = new HashMap<String, Composite>();
+
     if (!pipelines.isEmpty()) {
-      for (API.Pipeline result : pipelines) {
-        List<API.Stage> stages = result.getStagesList();
+      Composite stripComposite = withLayoutData( createComposite(folder, new FillLayout(SWT.HORIZONTAL)),
+          new GridData(SWT.FILL, SWT.TOP, true, false));
+      Label separator = withLayoutData( new Label(folder, SWT.SEPARATOR | SWT.HORIZONTAL),
+          new GridData(SWT.FILL, SWT.TOP, true, false));
+
+      StackLayout stageStack = new StackLayout();
+      Composite stageComposite = withLayoutData( createComposite(folder, stageStack),
+          new GridData(SWT.FILL, SWT.FILL, true, true));
+
+      for (int pipeIndex = 0; pipeIndex < pipelines.size(); pipeIndex++) {
+        List<API.Stage> stages = pipelines.get(pipeIndex).getStagesList();
+
+        for (int stageIndex = 0; stageIndex < stages.size(); stageIndex++) {
+          API.Stage stage = stages.get(stageIndex);
+          stageMap.put(stage.getDebugName(), createStage(stageComposite, stage));
+
+          Button stageButton = new Button(stripComposite, SWT.FLAT);
+          stageButton.setText(stage.getDebugName());
+          stageButton.addListener(SWT.Selection, e -> {
+            stageStack.topControl = stageMap.get(stage.getDebugName());
+            stageComposite.layout();
+            selectedStage = stage.getDebugName();
+            stripComposite.redraw();
+          });
+
+          stageButton.setBackground(getDisplay().getSystemColor(SWT.COLOR_LIST_BACKGROUND));
+          stageButton.setFont(theme.bigBoldFont());
+          stageButton.setToolTipText(stage.getStageName() + " (" + stage.getDebugName() + ")");
```

Comment:
No need for DebugName here, it is already in the button. Change to:
```stageButton.setToolTipText(stage.getStageName());```

Reply:
Ok, but just letting you know that the document and mockups include the debug name